patches/0002-Add-crypto-backend-foundation.patch

```diff
@@ -273,25 +277,41 @@ index 00000000000000..007d8070538247
 +)
 +
 +func init() {
-+	if v, _ := envGoFIPS(); v == "1" {
-+		if runtime.GOOS != "linux" {
-+			panic("FIPS mode requested (GOFIPS=1) but not supported on " + runtime.GOOS)
-+		} else if !iscgo {
-+			panic("FIPS mode requested (GOFIPS=1) but not supported with cgo disabled")
++	if v, r, ok := envGoFIPS(); ok && v == "1" {
++		if runtime.GOOS != "linux" && runtime.GOOS != "windows" {
++			panic("FIPS mode requested (" + r + ") but not supported on " + runtime.GOOS)
++		}
++		if runtime.GOOS == "linux" && !iscgo {
```

Comment:
Can this check be removed? We now panic at compile time when `opensslcrypto` is used without cgo instead of silently falling back to Go crypto, so it would be more informative to just say that no supported crypto backend is enabled.

If this check is removed, then we can also remove `iscgo.go` and `nocgo.go`.

Reply:
True, you really have to know what you're doing to force this error to pop up now:

```
$ CGO_ENABLED=0 go build -tags=goexperiment.opensslcrypto,requirefips,allow_missing_crypto_backend_fallback . && ./exp
panic: FIPS mode requested (requirefips tag set) but not supported with cgo disabled
```

I also moved the `runtime.GOOS` check into `if !Enabled` because I think it shows the intent better: it's just an explanation, and `Enabled` is really the only thing that really needs to be enforced.